### DISCUSSION TITLE
Mirth seemingly not converting reponses to JSON

### DISCUSSION THREAD

**Discussion by ghost** [nextgenhealthcare/connect#4578] [Q&A] [answered]
Mirth Version : 3.5.1

So I have this listener channel that receives HL7 messages, transforms each one in JSON, sends each one to an app, the app responts with a JSON and then the response transfer runs like two lines of javascript in case of an error.

I'm having trouble finding out why Mirth for some reason fails to parse the response.
The datatypes set on the channel are : Destination Hl72.X JSON and the response from this Destination is JSON JSON; 

Instead of getting just the JSON I'm getting this with Status: SENT
`<HttpResponse>
    <Status>HTTP/1.1 201 Created</Status>
    <Header>
        <Field>
            <Name>Transfer-Encoding</Name>
            <Value>chunked</Value>
        </Field>
        <Field>
            <Name>Server</Name>
            <Value>nginx/1.18.0</Value>
        </Field>
        <Field>
            <Name>X-Request-Id</Name>
            <Value>a18557a1-4faa-492c-82eb-84c3ad9dc270</Value>
        </Field>
        <Field>
            <Name>X-Content-Type-Options</Name>
            <Value>nosniff</Value>
        </Field>
        <Field>
            <Name>X-Runtime</Name>
            <Value>0.297039</Value>
        </Field>
        <Field>
            <Name>Connection</Name>
            <Value>keep-alive</Value>
        </Field>
        <Field>
            <Name>X-Download-Options</Name>
            <Value>noopen</Value>
        </Field>
        <Field>
            <Name>X-Permitted-Cross-Domain-Policies</Name>
            <Value>none</Value>
        </Field>
        <Field>
            <Name>Date</Name>
            <Value>Fri, 04 Jun 2021 09:19:21 GMT</Value>
        </Field>
        <Field>
            <Name>X-Frame-Options</Name>
            <Value>SAMEORIGIN</Value>
        </Field>
        <Field>
            <Name>X-Frame-Options</Name>
            <Value>SAMEORIGIN</Value>
        </Field>
        <Field>
            <Name>Referrer-Policy</Name>
            <Value>strict-origin-when-cross-origin</Value>
        </Field>
        <Field>
            <Name>Cache-Control</Name>
            <Value>max-age=0, private, must-revalidate</Value>
        </Field>
        <Field>
            <Name>ETag</Name>
            <Value>W/"aecd0f6d4e6f9c3ca287f3b433cdfd0d"</Value>
        </Field>
        <Field>
            <Name>X-XSS-Protection</Name>
            <Value>1; mode=block</Value>
        </Field>
        <Field>
            <Name>Content-Type</Name>
            <Value>application/json; charset=utf-8</Value>
        </Field>
    </Header>
    <Body multipart="no">{JSON_HERE}</Body>
</HttpResponse>
`

and the response transformer fails because I haven't used try

`SCRIPT SOURCE:	response
SOURCE CODE:	
421:     }
422: }
423: function doScript() {
424: msg = JSON.parse(connectorMessage.getResponseTransformedData());
425: function doTransform() {
426: 
427: 
428: if (response.getError() != null) {
429: responseStatus = ERROR;
430: }
LINE NUMBER:	426
DETAILS:	SyntaxError: Unexpected token: <
	at a70ce789-4c78-4c69-b558-4d71fc9ab234:426 (doScript)
	at a70ce789-4c78-4c69-b558-4d71fc9ab234:456
	at com.mirth.connect.server.transformers.JavaScriptResponseTransformer$ResponseTransformerTask.doCall(JavaScriptResponseTransformer.java:160)
	at com.mirth.connect.server.transformers.JavaScriptResponseTransformer$ResponseTransformerTask.doCall(JavaScriptResponseTransformer.java:118)
	at com.mirth.connect.server.util.javascript.JavaScriptTask.call(JavaScriptTask.java:113)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)`

Why does this happen ? Should I just set it to RAW ?

**Comment by rjkroll** [comment]
All it means is that the interpreter is expecting JavaScript or JSON and you are feeding it HTML/XML.  Check the src path to your JavaScript to make sure it is correct and that the syntax is also correct, including not misplacing any semicolons involved.  We would also need to see this script to see what you are trying to accomplish.

**Reply by ghost** [reply]
I'm not feeding it HTML/XML. If I recreate the call with Postman the response is just a JSON as in the {JSON_HERE} in the original post. That JSON gets automatically converted into that. 

![Screenshot_1](https://user-images.githubusercontent.com/84179902/120896363-755a3a80-c621-11eb-9a8f-9f9b6410258b.png)
![Screenshot_2](https://user-images.githubusercontent.com/84179902/120896370-7ee3a280-c621-11eb-89dc-c108651faec1.png)
![Screenshot_3](https://user-images.githubusercontent.com/84179902/120896397-9e7acb00-c621-11eb-9590-9a8478003227.png)

**Reply by tonygermano** [reply]
Check the Response Content setting on your HTTP Sender. You likely have it set to XML Body instead of Plain Body (in which case, you **are** feeding it xml.)

**Reply by ghost** [reply]
> Check the Response Content setting on your HTTP Sender. You likely have it set to XML Body instead of Plain Body (in which case, you **are** feeding it xml.)

 My HTTP Sender is an app. If I make the request with Postman or simply GET using the browser it will give me a JSON. How can I test this more profoundly ?

**Comment by pacmano1** [comment]
Perhaps post your channel export.

**Reply by ghost** [reply]
Here is the channel in question: https://pastebin.com/raw/j5qYq9Gh

**Comment by tonygermano** [comment, marked as answer]
From the channel you posted to pasetbin in https://github.com/nextgenhealthcare/connect/discussions/4578#discussioncomment-858575, in your `NewInvitation_ORM_NW` destination, you have Response Content set to XML Body, which is creating the XML you are seeing in your response transformer. I had suggested you check that setting https://github.com/nextgenhealthcare/connect/discussions/4578#discussioncomment-841186. Your `CancelInvitation_ORM_CA` HTTP Sender is not set that way, so I assume it is not having the same issue.

@rjkroll was correct that the error was indicating the channel was getting XML when it was expecting JSON. You could have set it to Raw like you proposed in the question to avoid the exception, but the setting on the destination would still have caused it to come in as XML.

**Comment by ghost** [comment]
Thank you all.
I can't believe I've missed the "Response Content" setting up until now.
